fastly/waf_rule_exclusion.go

```diff
@@ -0,0 +1,297 @@
+package fastly
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"reflect"
+	"strconv"
+	"time"
+
+	"github.com/google/jsonapi"
+)
+
+// WAFRuleExclusionType is used for reflection because JSONAPI wants to know what it's
+// decoding into.
+var WAFRuleExclusionType = reflect.TypeOf(new(WAFRuleExclusion))
+
+const (
+	// WAFRuleExclusionTypeRule is the type of WAF rule exclusions that excludes rules from the WAF based on certain conditions.
+	WAFRuleExclusionTypeRule = "rule"
+	// WAFRuleExclusionTypeWAF is the type of WAF rule exclusions that excludes WAF based on certain conditions.
+	WAFRuleExclusionTypeWAF = "waf"
+)
+
+// WAFRuleExclusion is the information about a WAF rule exclusion object.
+type WAFRuleExclusion struct {
+	ID            string     `jsonapi:"primary,waf_exclusion"`
+	Name          *string    `jsonapi:"attr,name"`
+	ExclusionType *string    `jsonapi:"attr,exclusion_type"`
+	Condition     *string    `jsonapi:"attr,condition"`
+	Number        *int       `jsonapi:"attr,number"`
+	Rules         []*WAFRule `jsonapi:"relation,waf_rules,omitempty"`
+	CreatedAt     *time.Time `jsonapi:"attr,created_at,iso8601,omitempty"`
+	UpdatedAt     *time.Time `jsonapi:"attr,updated_at,iso8601,omitempty"`
+}
+
+// WAFRuleExclusionResponse represents a list of rule exclusions - full response.
+type WAFRuleExclusionResponse struct {
+	Items []*WAFRuleExclusion
+	Info  infoResponse
+}
+
+// ListWAFRuleExclusionsInput used as input for listing a WAF's rule exclusions.
+type ListWAFRuleExclusionsInput struct {
+	// The Web Application Firewall's ID.
+	WAFID string
+	// The Web Application Firewall's version number.
+	WAFVersionNumber int
+	// Limit results to exclusions with the specified exclusions type.
+	FilterExclusionType *string
+	// Limit results to exclusions with the specified exclusion name.
+	FilterName *string
+	// Limit results to exclusions that represent the specified ModSecurity modsec_rule_id.
+	FilterModSedID *string
+	// Limit the number of returned pages.
+	PageSize *int
+	// Request a specific page of exclusions.
+	PageNumber *int
+	// Include relationships. Optional, comma-separated values. Permitted values: waf_rule_revision and waf_firewall_version.
+	Include *string
+}
+
+// ListAllWAFRuleExclusionsInput used as input for listing all WAF rule exclusions.
+type ListAllWAFRuleExclusionsInput struct {
+	// The Web Application Firewall's ID.
+	WAFID string
+	// The Web Application Firewall's version number.
+	WAFVersionNumber int
+	// Limit results to exclusions with the specified exclusions type.
+	FilterExclusionType *string
+	// Limit results to exclusions with the specified exclusion name.
+	FilterName *string
+	// Limit results to exclusions that represent the specified ModSecurity modsec_rule_id.
+	FilterModSedID *string
+	// Include relationships. Optional, comma-separated values. Permitted values: waf_rule_revision and waf_firewall_version.
+	Include *string
```

Comment:
Same as above, this should be a `[]string`.

Reply:
Updated to slice as well